src/api/roles/coach/coach.py

```
@@ -1204,32 +1353,69 @@ def get_client_progress_pictures(
         .where(ClientTelemetry.client_id == client_id)
         .order_by(CompletedMealActivity.id.desc())
     )
-    return db.exec(query.offset(pagination.skip).limit(pagination.limit)).all()
+    rows = db.exec(query.offset(pagination.skip).limit(pagination.limit)).all()
+
+    parsed_date = None
+    if on_date:
+        try:
+            parsed_date = datetime.strptime(on_date, "%Y-%m-%d").date()
+        except ValueError:
+            raise HTTPException(400, "on_date must be YYYY-MM-DD")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)